src/flp/types/dp.rs

```diff
@@ -0,0 +1,331 @@
+use crate::dp::{distributions::PureDpDiscreteLaplace, DifferentialPrivacyStrategy};
+use crate::dp::{DifferentialPrivacyDistribution, DpError};
+use crate::field::{FftFriendlyFieldElement, Field128, Field64};
+use crate::flp::gadgets::{Mul, ParallelSumGadget};
+use crate::flp::types::{Histogram, SumVec};
+use crate::flp::{FlpError, TypeWithNoise};
+use crate::vdaf::xof::SeedStreamTurboShake128;
+use num_bigint::{BigInt, BigUint, TryFromBigIntError};
+use num_integer::Integer;
+use rand::{distributions::Distribution, Rng, SeedableRng};
+
+// TODO(#1071): This is implemented for the concrete fields `Field64` and `Field128` in order to
+// avoid imposing the `BigInt: From<F::Integer>` bound on all callers. In the future, we may want to
+// impose this bound on `FieldElementWithInteger`, which would allow simple blanket implementations
+// here.
+impl<S> TypeWithNoise<PureDpDiscreteLaplace> for SumVec<Field64, S>
+where
+    S: ParallelSumGadget<Field64, Mul<Field64>> + Eq + 'static,
+{
+    fn add_noise_to_result(
+        &self,
+        dp_strategy: &PureDpDiscreteLaplace,
+        agg_result: &mut [Self::Field],
+        _num_measurements: usize,
+    ) -> Result<(), FlpError> {
+        self.add_noise(
+            dp_strategy,
+            agg_result,
+            &mut SeedStreamTurboShake128::from_entropy(),
+        )
+    }
+}
+
+impl<S> TypeWithNoise<PureDpDiscreteLaplace> for SumVec<Field128, S>
+where
+    S: ParallelSumGadget<Field128, Mul<Field128>> + Eq + 'static,
+{
+    fn add_noise_to_result(
+        &self,
+        dp_strategy: &PureDpDiscreteLaplace,
+        agg_result: &mut [Self::Field],
+        _num_measurements: usize,
+    ) -> Result<(), FlpError> {
+        self.add_noise(
+            dp_strategy,
+            agg_result,
+            &mut SeedStreamTurboShake128::from_entropy(),
+        )
+    }
+}
+
+impl<F, S> SumVec<F, S>
+where
+    F: FftFriendlyFieldElement,
+    BigInt: From<F::Integer>,
+    F::Integer: TryFrom<BigInt, Error = TryFromBigIntError<BigInt>>,
+{
+    fn add_noise<R>(
+        &self,
+        dp_strategy: &PureDpDiscreteLaplace,
+        agg_result: &mut [F],
+        rng: &mut R,
+    ) -> Result<(), FlpError>
+    where
+        R: Rng,
+    {
+        // Compute the l1-sensitivity of the aggregation function (assuming the substitution-DP
```

Comment:
Same comment as above
```suggestion
        // Compute the L1-sensitivity of the aggregation function (assuming the substitution-DP
```